E621Client/Converters/PostImageConverter.cs

```diff
@@ -10,15 +12,30 @@ namespace Noppes.E621.Converters
     /// </summary>
     internal class PostImageConverter<TPostImage> : JsonConverter<TPostImage?> where TPostImage : PostImage
```

Comment:
Wouldn't the solution to your problem be to just remove the whole `PostImageConverter`? That way we don't ignore these fields for deleted posts.